plugins/pulp_rpm/plugins/profilers/yum.py

```diff
@@ -87,12 +87,24 @@ def install_units(consumer, units, options, config, conduit):
                                        the specified errata
         """
         translated_units = []
+        available_rpm_nevras = None
+
         for unit in units:
             if unit['type_id'] == TYPE_ID_RPM:
                 translated_units.append(unit)
             elif unit['type_id'] == TYPE_ID_ERRATA:
+                # Populate the available RPM cache if needed. Check
+                # specifically for "None" since [] could be valid in the case
+                # of no available RPMs
+                if available_rpm_nevras is None:
+                    available_rpm_nevras = []
+                    for bound_repo_id in conduit.get_bindings(consumer.id):
+                        for r in conduit.get_repo_units(bound_repo_id, TYPE_ID_RPM):
```

Comment:
I am updating now to do this :racehorse:. It also removes the need for the cache which is nice.